Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -422,7 +422,26 @@ - (void)configureViews
         // Do nothing.
     } else if (shouldFooterOverlayMedia) {
         OWSAssert(bodyMediaView);
-        [self.footerView configureWithConversationViewItem:self.viewItem hasShadows:YES];
+
+        CGFloat maxGradientHeight = 48.f;
+        CAGradientLayer *gradientLayer = [CAGradientLayer new];
+        gradientLayer.colors = @[
+            (id)[UIColor colorWithWhite:0.f alpha:0.f].CGColor,
+            (id)[UIColor colorWithWhite:0.f alpha:0.4f].CGColor,
+        ];
+        OWSLayerView *gradientView =
```

Comment:
Did you take a look at using the existing GradientView class?

Reply:
The layout of this gradient is slightly complicated.  IMO it's simpler (and more readable) to express that using a 4-line block than ~5 layout constraints.